metriq_gym/run.py

```diff
@@ -9,17 +8,29 @@
 from metriq_gym.job_type import JobType
 from metriq_gym.schema_validator import load_and_validate
 
-
 logging.basicConfig(level=logging.INFO, format="%(asctime)s - %(levelname)s - %(message)s")
 
 
 def main():
     load_dotenv()
     args = parse_arguments()
-    params = load_and_validate(args.input_file)
     job_manager = JobManager(args.jobs_file)
 
-    HANDLERS[JobType(params["benchmark_name"])](args, params, job_manager).dispatch_handler()
+    if args.action == "dispatch":
+        params = load_and_validate(args.input_file)
+
+        handler = HANDLERS[JobType(params["benchmark_name"])](args, params, job_manager)
+        handler.dispatch_handler()
+    elif args.action == "poll":
+        job_id = args.job_id
+        job = job_manager.get_job(job_id)
+        job_type = job["job_type"]
+
+        handler = HANDLERS[job_type](args, None, job_manager)
```

Comment:
you still need to wrap `job_type` into `JobType(·)` if you want to use it as key of HANDLERS.

Reply:
Good catch, we will want that. Done in eed1ecd.